pandas/_libs/src/ujson/python/objToJSON.c

```diff
@@ -2126,6 +2133,19 @@ double Object_getDoubleValue(JSOBJ Py_UNUSED(obj), JSONTypeContext *tc) {
     return GET_TC(tc)->doubleValue;
 }
 
+const char *Object_getBigNumStringValue(JSOBJ obj, JSONTypeContext *tc, 
+                                    size_t *_outLen) {
+    PyObject* repr = PyObject_Str(obj);
+    const char *str = PyUnicode_AsUTF8AndSize(repr, (Py_ssize_t *) _outLen);
+    char* bytes = malloc(*_outLen);
```

Comment:
Knowing there's some strange results still outstanding, you *may* need to allocate one extra byte here for a null terminator